Sources/MapboxDirections/Incident.swift

```diff
@@ -21,46 +20,156 @@ public struct Incident: Codable, Equatable {
         case geometryIndexStart = "geometry_index_start"
         case geometryIndexEnd = "geometry_index_end"
     }
-
-    var identifier: String
-    var type: String
-    var description: String
-    var creationTime: String
-    var startTime: String
-    var endTime: String
-    var impact: String?
-    var subtype: String?
-    var subtypeDescription: String?
-    var alertCodes: [Int]
-    var lanesBlocked: [Int]
-    var geometryIndexStart: Int
-    var geometryIndexEnd: Int
+    
+    /// Defines known types of incidents.
+    ///
+    /// Each incident may or may not have specific set of data, depending on it's `kind`
+    public enum Kind: String {
+        /// Accident
+        case accident = "accident"
+        /// Congestion
+        case congestion = "congestion"
+        /// Construction
+        case construction = "construction"
+        /// Disabled vehicle
+        case disabledVehicle = "disabled_vehicle"
+        /// Lane restriction
+        case laneRestriction = "lane_restriction"
+        /// Mass transit
+        case massTransit = "mass_transit"
+        /// Miscellaneous
+        case miscellaneous = "miscellaneous"
+        /// Other news
+        case otherNews = "other_news"
+        /// Planned event
+        case plannedEvent = "planned_event"
+        /// Road closure
+        case roadClosure = "road_closure"
+        /// Road hazard
+        case roadHazard = "road_hazard"
+        /// Weather
+        case weather = "weather"
+    }
+    
+    /// Incident identifier
+    public var identifier: String
+    /// The kind of an incident
+    ///
+    /// This value is set to `nil` if `kind` value is not supported.
+    public var kind: Kind? {
+        return Kind(rawValue: rawKind)
+    }
+    var rawKind: String
```

Comment:
Something for consideration before the final release: now that we’re writing in pure Swift, we could consolidate these two properties into a single `kind` property of type `Kind`, but one of the `Kind` cases could have an associated value with the raw type. This would in fact be an option for any enumeration that the Directions API reserves the right to extend, other than option sets.